tests/test_esm_parser/test_provenance.py

```diff
@@ -164,122 +150,172 @@ def test_extract_dict():
-                    'line': 14,
-                    'col': 18,
-                    'yaml_file': file_path,
-                    'category': 'None'}, {
-                        'line': 14,
-                        'col': 21,
-                        'yaml_file': file_path,
-                        'category': 'None'}, {
-                        'line': 14,
-                        'col': 24,
-                        'yaml_file': file_path,
-                        'category': 'None'}],
-                'my_bolean': {
-                        'line': 16,
-                        'col': 13,
-                        'yaml_file': file_path,
-                        'category': 'None'},
-                'my_int': {
-                        'line': 17,
-                        'col': 10,
-                        'yaml_file': file_path,
-                        'category': 'None'},
-                'my_int2': {
-                        'line': 18,
-                        'col': 11,
-                        'yaml_file': file_path,
-                        'category': 'None'},
-                'list_with_dict_inside': [{
-                    'line': 20,
-                    'col': 4,
-                    'yaml_file': file_path,
-                    'category': 'None'}, {
-                        'line': 21,
-                        'col': 4,
-                        'yaml_file': file_path,
-                        'category': 'None'}, {
-                        'my_dict': {
-                            'foo': [{
-                                'line': 24,
-                                'col': 10,
-                                'yaml_file': file_path,
-                                'category': 'None'}, {
-                                    'line': 25,
-                                    'col': 10,
-                                    'yaml_file': file_path,
-                                    'category': 'None'}, {
-                                    'my_dict': {
-                                        'foo': {
-                                            'line': 27,
-                                            'col': 17,
-                                            'yaml_file': file_path,
-                                            'category': 'None'}
-                                    }}]}}]}}
+    file_path = pathlib.Path(str(pathlib.Path(f"{os.environ['GITHUB_WORKSPACE']}/tests/test_esm_parser/example.yaml")))
+    check_prov = [{'line': 15, 'col': 19, 'yaml_file': str(pathlib.Path(f"{os.environ['GITHUB_WORKSPACE']}/tests/test_esm_parser/example.yaml").resolve()), 'category': 'runscript'}, {'line': 15, 'col': 22, 'yaml_file': str(pathlib.Path(f"{os.environ['GITHUB_WORKSPACE']}/tests/test_esm_parser/example.yaml").resolve()), 'category': 'runscript'}, {'line': 15, 'col': 25, 'yaml_file': str(pathlib.Path(f"{os.environ['GITHUB_WORKSPACE']}/tests/test_esm_parser/example.yaml").resolve()), 'category': 'runscript'}]
```

Comment:
Potentially the same problem for `os.environ['GITHUB_WORKSPACE']` when running the tests locally. It's probably worth it to define a variable outside of the functions that define the path to the file you are trying to reach depending on an `if os.environ['GITHUB_WORKSPACE']`